plugin/lighthouse/handler.go

```diff
@@ -111,6 +111,20 @@ func (lh *Lighthouse) emptyResponse(state request.Request) (int, error) {
 	return dns.RcodeSuccess, nil
 }
 
+func (lh *Lighthouse) getClusterIpForSvc(pReq recordRequest) (ip string, found bool) {
+	localClusterID := lh.clusterStatus.LocalClusterID()
+
+	ip, found, isLocal := lh.serviceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, localClusterID, lh.clusterStatus.IsConnected,
+		lh.endpointsStatus.IsHealthy)
+
+	getLocal := isLocal || (pReq.cluster != "" && pReq.cluster == localClusterID)
```

Comment:
The OR clause isn't needed here with the suggested change  `serviceImports.GetIP`.